server/services/store/mattermostauthlayer/mattermostauthlayer.go

```diff
@@ -428,6 +430,92 @@ func (s *MattermostAuthLayer) GetUserWorkspaces(userID string) ([]model.UserWork
 	return s.userWorkspacesFromRows(rows)
 }
 
+func (s *MattermostAuthLayer) GetUserWorkspacesInTeam(userID string, teamID string) ([]model.UserWorkspace, error) {
+	var query sq.SelectBuilder
+
+	var nonTemplateFilter string
+
+	switch s.dbType {
+	case mysqlDBType:
+		nonTemplateFilter = nonTemplateFilterMySQL
+	case postgresDBType:
+		nonTemplateFilter = nonTemplateFilterPostgres
+	default:
+		return nil, fmt.Errorf("GetUserWorkspaces - %w", errUnsupportedDatabaseError)
+	}
+
+	query = s.getQueryBuilder().
+		Select("Channels.ID", "Channels.DisplayName", "COUNT(focalboard_blocks.id)", "Channels.Type", "Channels.Name").
+		From("ChannelMembers").
+		// select channels without a corresponding workspace
+		LeftJoin(
+			"focalboard_blocks ON focalboard_blocks.workspace_id = ChannelMembers.ChannelId AND "+
+				"focalboard_blocks.type = 'board' AND "+
+				nonTemplateFilter,
+		).
+		Join("Channels ON ChannelMembers.ChannelId = Channels.Id").
+		Where(sq.Eq{"ChannelMembers.UserId": userID}).
+		Where(sq.Eq{"Channels.TeamID": teamID}).
+		GroupBy("Channels.Id", "Channels.DisplayName")
+
+	rows, err := query.Query()
+	if err != nil {
+		s.logger.Error("ERROR GetUserWorkspaces", mlog.Err(err))
+		return nil, err
+	}
+
+	defer s.CloseRows(rows)
+
+	accessibleWorkspaces, err := getPublicWorkspacesInTeam(teamID, s)
```

Comment:
We can make it better that way, surely. However, here are some of the catches that made me decide against it.

1) In this version of code, there aren't tests for `plugin-version` of focalboard. So the blocks defined here should be as clean as possible.
2) Since the norm is to use Squirrel in focalboard, the only way we can `UNION` is to do something like [this](https://github.com/Masterminds/squirrel/issues/20#issuecomment-508680438), which I've added in my most recent commit.

This way seemed cleaner to me than use of `suffixExpr` because of the parameter counting issues with squirrel.

Please let me know what you think. @mkraft